aten/src/ATen/native/SpectralOps.cpp

```diff
@@ -136,16 +145,24 @@ Tensor fft_r2c(Tensor input, c10::optional<int64_t> n_opt,
   }
 
   const auto norm = norm_from_string(norm_str, forward);
-  auto out = at::_fft_r2c(input, dim, static_cast<int64_t>(norm), onesided);
+
+  Tensor ret;
+  if (out.defined() && forward) {
+    ret = at::_fft_r2c_out(out, input, dim, static_cast<int64_t>(norm), onesided);
+  } else {
+    ret = at::_fft_r2c(input, dim, static_cast<int64_t>(norm), onesided);
+  }
+
   if (!forward) {
     // FIXME: _fft_r2c doesn't support native r2c IFFT
-    out = at::conj(out);
+    return out.defined() ? at::conj_out(out, ret) : at::conj(ret);
```

Comment:
In the case that out is defined, ret == out, right? Would it be more natural to perform the conj inplace instead of using its out variant? 

This question is mostly academic since conj doesn't have an inplace variant. cc @anjali411 we should probably add an inplace `conj_` for consistency with our other operators?

Reply:
@mruberry yeah I agree I don't see any reason to not add an in-place version of conj